paperqa/agents/tools.py

```diff
@@ -190,19 +190,29 @@ async def gather_evidence(self, question: str, state: EnvironmentState) -> str:
         if not state.docs.docs:
             raise EmptyDocsError("Not gathering evidence due to having no papers.")
 
+        if f"{self.TOOL_FN_NAME}_initialized" in self.settings.callbacks:
+            callback_list = self.settings.callbacks[f"{self.TOOL_FN_NAME}_initialized"]
+            for callback in callback_list:
+                await callback(state)
+
         logger.info(f"{self.TOOL_FN_NAME} starting for question {question!r}.")
         original_question = state.answer.question
         try:
             # Swap out the question with the more specific question
             # TODO: remove this swap, as it prevents us from supporting parallel calls
             state.answer.question = question
             l0 = len(state.answer.contexts)
+
+            aget_evidence_callbacks = self.settings.callbacks.get(
+                f"{self.TOOL_FN_NAME}_aget_evidence", None
+            )
             # TODO: refactor answer out of this...
             state.answer = await state.docs.aget_evidence(
                 query=state.answer,
                 settings=self.settings,
                 embedding_model=self.embedding_model,
                 summary_llm_model=self.summary_llm_model,
+                callbacks=aget_evidence_callbacks,
```

Comment:
style nitpick, but you can probably just insert the expression directly rather than making a variable via `aget_evidence_callbacks`. Also the get("my key", None) is the default, so we usually don't specify it.

Reply:
ah, good points! will do!